mysql-test/suite/innodb/t/instant_alter_charset.test

```diff
@@ -0,0 +1,527 @@
+
+  eval alter table tmp
+    change a a varchar(50) charset $to_charset collate $to_collate,
+    modify b varchar(50) charset $to_charset collate $to_collate,
+    algorithm=instant;
+
+  check table tmp;
+
+  drop table tmp;
+
+  inc $counter;
+}
+
+drop table fully_compatible;
+
+
+create table compatible_without_index (
+  id int auto_increment unique key,
+  from_charset char(255),
+  from_collate char(255),
+  to_charset char(255),
+  to_collate char(255)
+);
+
+insert into compatible_without_index (from_charset, from_collate, to_charset, to_collate) values
+  ('ascii', 'ascii_general_ci',       'utf8mb3', 'utf8mb3_swedish_ci'),
+  ('ascii', 'ascii_bin',              'latin1', 'latin1_swedish_ci'),
+  ('ascii', 'ascii_general_nopad_ci', 'latin1', 'latin1_swedish_ci'),
+  ('ascii', 'ascii_nopad_bin',        'latin1', 'latin1_swedish_ci'),
+
+  ('ascii', 'ascii_general_ci',       'koi8u', 'koi8u_bin'),
+  ('ascii', 'ascii_general_nopad_ci', 'koi8u', 'koi8u_bin'),
+  ('ascii', 'ascii_nopad_bin',        'koi8u', 'koi8u_bin'),
+
+  ('ascii', 'ascii_general_ci',       'latin1', 'latin1_swedish_ci'),
+  ('ascii', 'ascii_bin',              'utf8mb3', 'utf8mb3_swedish_ci'),
+  ('ascii', 'ascii_general_nopad_ci', 'utf8mb3', 'utf8mb3_swedish_ci'),
+  ('ascii', 'ascii_nopad_bin',        'utf8mb3', 'utf8mb3_swedish_ci'),
+
+  ('ascii', 'ascii_general_ci',       'utf8mb4', 'utf8mb4_danish_ci'),
+  ('ascii', 'ascii_bin',              'utf8mb4', 'utf8mb4_danish_ci'),
+  ('ascii', 'ascii_general_nopad_ci', 'utf8mb4', 'utf8mb4_danish_ci'),
+  ('ascii', 'ascii_nopad_bin',        'utf8mb4', 'utf8mb4_danish_ci'),
+
+  ('utf8mb3', 'utf8mb3_general_ci',       'utf8mb4', 'utf8mb4_vietnamese_ci'),
+  ('utf8mb3', 'utf8mb3_bin',              'utf8mb4', 'utf8mb4_vietnamese_ci'),
+  ('utf8mb3', 'utf8mb3_general_nopad_ci', 'utf8mb4', 'utf8mb4_vietnamese_ci'),
+  ('utf8mb3', 'utf8mb3_nopad_bin',        'utf8mb4', 'utf8mb4_vietnamese_ci'),
+
+  ('ascii',  'ascii_general_ci',     'gbk',  'gbk_chinese_ci'),
+  ('ascii',  'ascii_general_ci',     'gbk',  'gbk_chinese_nopad_ci'),
+
+  ('ascii',   'ascii_general_ci',      'ascii',   'ascii_bin'),
+  ('utf8mb3', 'utf8mb3_roman_ci',      'utf8mb3', 'utf8mb3_lithuanian_ci'),
+  ('utf8mb4', 'utf8mb4_thai_520_w2',   'utf8mb4', 'utf8mb4_persian_ci'),
+  ('utf8mb3', 'utf8mb3_myanmar_ci',    'utf8mb4', 'utf8mb4_german2_ci'),
+  ('utf8mb3', 'utf8mb3_general_ci',    'utf8mb3', 'utf8mb3_unicode_ci'),
+  ('latin1',  'latin1_general_cs',     'latin1',  'latin1_general_ci'),
+  ('ascii',   'ascii_general_ci',      'ujis',    'ujis_japanese_ci'),
+  ('ascii',   'ascii_general_ci',      'sjis',    'sjis_japanese_ci'),
+  ('ascii',   'ascii_general_ci',      'big5',    'big5_chinese_ci'),
+  ('ascii',   'ascii_general_ci',      'latin2',  'latin2_croatian_ci'),
+  ('ascii',   'ascii_general_ci',      'latin7',  'latin7_estonian_cs'),
+  ('utf16',   'utf16_general_ci',      'utf16',   'utf16_german2_ci')
+;
```

Comment:
These do not seem to be all the combinations. A comment about this would be good.